imap_data_access/file_validation.py

```diff
@@ -0,0 +1,264 @@
+"""Methods for managing and validating filenames and filepaths"""
+from __future__ import annotations
+
+import re
+
+from datetime import datetime
+from pathlib import Path
+
+VALID_INSTRUMENTS = {
+    "codice",
+    "glows",
+    "hit",
+    "hi-45",
+    "hi-90",
+    "idex",
+    "lo",
+    "mag",
+    "swapi",
+    "swe",
+    "ultra-45",
+    "ultra-90",
+}
+
+VALID_DATALEVELS = {"l0", "l1", "l1a", "l1b", "l1c", "l1d", "l2"}
+
+VALID_FILE_EXTENSION = {"pkts", "cdf"}
+
+FILENAME_CONVENTION = (
+    "<mission>_<instrument>_<datalevel>_<descriptor>_"
+    "<startdate>_<enddate>_<version>.<extension>"
+)
```

Comment:
This is nitpicky and I don't have strong opinions on this, but I wonder if these should be defined in ``__init__.py`` so that other repos can import like this

```
import imap_data_access
imap_data_access.VALID_INSTRUMENTS
```

instead of 

```
from imap_data_access import file_validation
file_validation.VALID_INSTRUMENTS
```

I can see a case where a developer just needs a list of valid instruments without having to validate a filename.

Reply:
Yeah, I like that. I'll move them. The easier it is to access this stuff, the less likely it is that someone will re-define it, lol